runtime/src/accounts.rs

```diff
@@ -42,6 +42,11 @@ pub struct Accounts {
     /// and number of locks. On commit_credits(), we do a take() on the option so that the hashmap
     /// is no longer available to be written to.
     credit_only_account_locks: Arc<RwLock<Option<HashMap<Pubkey, CreditOnlyLock>>>>,
+
+    /// Set the rent owed by credit only account in the tx
```

Comment:
great comment, but I'm not sure I follow.  can you expand upon this?

Reply:
So, reason why we can't just use `credit_only_locks` is, it is being persisted regardless of tx fails or succeeds. 
Which leads to a scenario, where a tx fails, and rent is not charged for credit debit account, but it is charged for credit only.

To prevent that discrepancy, we need to store credit only rent debtors independently.

Reply:
I think it's fine to collect rent when rent is due.  We'd considered having a "noop" instruction that would help take care of this.  Looks like you've found one.

Reply:
But, this would be inconsistent, no? I mean, we will be charging rent to credit only, but skip for credit-debit. 

In my opinion, let's leave this as it is, and will open another issue to address `noop` instruction.

Reply:
Ok, works for me.  There's no way to combine those data structures into one?

Inconsistency isn't the concern I have, though: the main reason for keeping it like this is that unless the result of a TX can_commit() the bank hash shouldn't change (there's a lot of code that depends on this, e.g. voting).  Can you add a test for this?

Reply:
sure. Will add.

Reply:
added.

Reply:
still no resolution on whether you can combine these two HashXXX structures to collapse the lookups?

Reply:
done. added the design